src/storage/table_iterator_test.cc

```diff
@@ -0,0 +1,353 @@
+/*
+ * Copyright 2021 4Paradigm
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "codec/schema_codec.h"
+#include "codec/sdk_codec.h"
+#include "common/timer.h"
+#include "gtest/gtest.h"
+#include "storage/mem_table.h"
+#include "storage/disk_table.h"
+#include "base/file_util.h"
+
+
+DECLARE_string(ssd_root_path);
+DECLARE_string(hdd_root_path);
+
+namespace openmldb {
+namespace storage {
+
+std::atomic<int32_t> counter(0);
+
+inline uint32_t GenRand() {
+    srand((unsigned)time(NULL));
+    return rand() % 10000000 + 1;
+}
+
+void RemoveData(const std::string& path) {
+    ::openmldb::base::RemoveDir(path + "/data");
+    ::openmldb::base::RemoveDir(path);
+    ::openmldb::base::RemoveDir(FLAGS_hdd_root_path);
+    ::openmldb::base::RemoveDir(FLAGS_ssd_root_path);
+}
+
+Table* CreateTable(const std::string& name, uint32_t id, uint32_t pid, uint32_t seg_cnt,
+                   const std::map<std::string, uint32_t>& mapping, uint64_t ttl, ::openmldb::type::TTLType ttl_type,
+                   const std::string& table_path, ::openmldb::common::StorageMode storage_mode) {
+    if (storage_mode == ::openmldb::common::StorageMode::kMemory) {
+        return new MemTable(name, id, pid, seg_cnt, mapping, ttl, ttl_type);
+    } else {
+        return new DiskTable(name, id, pid, mapping, ttl, ttl_type, storage_mode, table_path);
+    }
+}
+
+Table* CreateTable(const ::openmldb::api::TableMeta& table_meta, const std::string& table_path) {
+    if (table_meta.storage_mode() == ::openmldb::common::StorageMode::kMemory) {
+        return new MemTable(table_meta);
+    } else {
+        return new DiskTable(table_meta, table_path);
+    }
+}
+
+inline std::string GetDBPath(const std::string& root_path, uint32_t tid, uint32_t pid) {
+    return root_path + "/" + std::to_string(tid) + "_" + std::to_string(pid);
+}
+
+using ::openmldb::codec::SchemaCodec;
+
+class DiskTestEnvironment : public ::testing::Environment{
+    virtual void TearDown() {
+        for (int i = 1; i <= counter; i++) {
+            std::string path = FLAGS_hdd_root_path + "/" + std::to_string(i) +  "_1";
+            RemoveData(path);
+        }
+    }
+};
+
+class TableIteratorTest : public ::testing::TestWithParam<::openmldb::common::StorageMode> {
```

Comment:
can rename `mem_table_iterator_test.cc` to this file

Reply:
Tests in `memtable_iterator_test.cc` are included in `table_iterator_test.cc`. Remove `memtable_iterator_test.cc` ?

Reply:
Yes. better `git mv memtable_iterator_test.cc table_iterator_test.cc`. so the git history is kept.